mgmtsystem_audit/models/mgmtsystem_audit.py

```diff
@@ -20,163 +20,265 @@
-                context=context
-            )
-        return super(mgmtsystem_audit, self).message_auto_subscribe(
-            cr, uid, ids, updated_fields, context=context, values=values
+        self.ensure_one()
+        user_ids = [self.user_id.id]
+        user_ids += [a.id for a in self.auditor_user_ids]
+        user_ids += [a.id for a in self.auditee_user_ids]
+
+        self.message_subscribe_users(user_ids=user_ids, subtype_ids=None)
+
+        return super(MgmtSystemAudit, self).message_auto_subscribe(
+            updated_fields=updated_fields,
+            values=values
         )
 
-    def get_audit_url(self, cr, uid, ids, context=None):
+    def get_action_url(self):
         """
         Return a short link to the audit form view
         eg. http://localhost:8069/?db=prod#id=1&model=mgmtsystem.audit
         """
-        assert len(ids) == 1
-        audit = self.browse(cr, uid, ids[0], context=context)
-        base_url = self.pool.get('ir.config_parameter').get_param(
-            cr, uid, 'web.base.url', default='http://localhost:8069',
-            context=context,
+
+        base_url = self.env['ir.config_parameter'].get_param(
+            'web.base.url',
+            default='http://localhost:8069'
         )
-        query = {'db': cr.dbname}
-        fragment = {'id': audit.id, 'model': self._name}
-        return urljoin(
-            base_url, "?%s#%s" %
-            (urlencode(query), urlencode(fragment))
+        url = ('{}/web#db={}&id={}&model={}').format(
+            base_url,
+            self.env.cr.dbname,
+            self.id,
+            self._name
         )
+        return url
+
+    def get_lines_by_procedure(self):
+        p = []
+        for l in self.line_ids:
+            if l.procedure_id.id:
+                proc_nm = self.pool.get('document.page').read(
+                    self.env.cr, self.env.uid, l.procedure_id.id, ['name']
+                )
+                procedure_name = proc_nm['name']
+            else:
+                procedure_name = _('Undefined')
+
+            p.append({"id": l.id,
+                      "procedure": procedure_name,
+                      "name": l.name,
+                      "yes_no": "Yes / No"})
+        p = sorted(p, key=lambda k: k["procedure"])
+        proc_line = False
+        q = []
+        proc_name = ''
+        for i in range(len(p)):
+            if proc_name != p[i]['procedure']:
+                proc_line = True
+            if proc_line:
+                q.append({"id": p[i]['id'],
+                          "procedure": p[i]['procedure'],
+                          "name": "",
+                          "yes_no": ""})
+                proc_line = False
+                proc_name = p[i]['procedure']
+            q.append({"id": p[i]['id'],
+                      "procedure": "",
+                      "name": p[i]['name'],
+                      "yes_no": "Yes / No"})
+        return q
```

Comment:
I searched, but didn't find how to use read_group() in qweb. Please can you provide me some tips or link?